lib/pbench/cli/agent/commands/tools/list.py

```diff
@@ -29,16 +29,21 @@ def print_results(toolinfo: dict, with_option: bool) -> bool:
         """
         printed = False
         for group, gval in sorted(toolinfo.items()):
-            for host, tools in sorted(gval.items()):
+            for host, hostitems in sorted(gval.items()):
+                label = toolinfo[group][host]["label"]
+                host_string = (
+                    f"host: {host}" if not label else f"host: {host}, label: {label}"
+                )
```

Comment:
Alternately,
```python
                host_string = f"host: {host}" + f", label: {label}" if label else ""
```
or
```python
                host_string = f"host: {host}{f', label: {label}' if label else ''}"
```
(which I think Black will leave on one line...).

Reply:
The last one stopped me for a few seconds, so I think I'll avoid it. I'll see if Black treats the first one better.

Reply:
> The last one stopped me for a few seconds, so I think I'll avoid it.

Good for you!  While I very much appreciate having f-strings in my toolbox, I think they are way over-used.  For instance, the first of the two alternatives above works just as nicely without the first f-string, and it's not hard to remove the second, although it does require some parentheses:

```python
                host_string = "host: " + host + (", label: " + label) if label else ""
```

I imagine that the concatenation is faster than the interpolation (but, perhaps it depends...), so it all comes down to readability.  I think that using the second f-string is better than using the parenthesized expression...but I'm not sure about whether the first one is more readable than the concatenation or not (I suspect it depends on who's doing the reading and what they are used to seeing).

Reply:
It needs parens:

```python
                          host_string = f"host: {host}" + (f", label: {label}" if label else "")
```

Reply:
My bad!  (Apologies....)

Adding the parens makes the nested f-string look less bad in comparison....  😉